File: src/Gostio.Services/Database/Entities/NewsItem.cs
namespace Gostio.Services.Database.Entities;

public class NewsItem
{
    public int Id { get; set; }

    public int CreatedByUserId { get; set; }

    public User CreatedByUser { get; set; } = null!;

    public string Title { get; set; } = null!;

    public string Body { get; set; } = null!;

    public byte[] Image { get; set; } = null!;

    public string ImageContentType { get; set; } = null!;

    public DateTime PublishedAt { get; set; }

    public DateTime? ModifiedAt { get; set; }
}

File: apps/gostio_desktop/lib/features/news/data/news_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'news_item.g.dart';

// No list carries bytes, so a row names where its picture is read from.
@JsonSerializable(createToJson: false)
class NewsItem {
  const NewsItem({
    required this.id,
    required this.title,
    required this.body,
    required this.imageContentType,
    required this.authorId,
    required this.authorName,
    required this.publishedAt,
    this.modifiedAt,
  });

  factory NewsItem.fromJson(Map<String, dynamic> json) =>
      _$NewsItemFromJson(json);

  final int id;
  final String title;
  final String body;
  final String imageContentType;
  final int authorId;
  final String authorName;
  final DateTime publishedAt;
  final DateTime? modifiedAt;

  String get imagePath => '/news/$id/image';
}
